spec.bs

```diff
@@ -3786,20 +3795,67 @@ The <dfn for=Navigator method>getInterestGroupAdAuctionData(|configIDL|)</dfn> m
-     |igPAggCoordinatorMap|. The serialization method may follow that described in
-     [Section 2.2.4 of Bidding and Auction Services](https://privacysandbox.github.io/draft-ietf-bidding-and-auction-services/draft-ietf-bidding-and-auction-services.html#name-generating-a-request).
-  1. Set |result|["{{AdAuctionData/request}}"] to |requestBlob|.
+  1. [=list/For each=] |config| of |configs|:
+    1. Let |seller| be |config|'s [=auction data config/seller=].
+    1. If |config|'s [=auction data config/coordinator=] is not one of the [=implementation-defined=]
+       coordinators supported by this [=user agent=]:
+      1. Let |result| be a new [=auction data per seller result=] with the following [=struct/items=]:
+        : [=auction data per seller result/seller=]
+        :: |seller|
+        : [=auction data per seller result/error=]
+        :: "invalid coordinator"
+      1. [=list/Append=] |result| to |results|.
+      1. [=iteration/Continue=].
+    1. Let |keyInfo| be the result of [=looking up the server encryption key=]
+       with |config|'s [=auction data config/seller=] and |seller|'s [=auction data config/coordinator=].
+    1. If |keyInfo| is failure:
+          1. Let |result| be a new [=auction data per seller result=] with the following [=struct/items=]:
+            : [=auction data per seller result/seller=]
+            :: |seller|
+            : [=auction data per seller result/error=]
+            :: "key lookup failed"
+        1. [=list/Append=] |result| to |results|.
+        1. [=iteration/Continue=].
+    1. Let (|key|, |keyId|) be |keyInfo|.
+    1. Set |config|'s [=auction data config/encryption key=] to |key|.
+    1. Set |config|'s [=auction data config/encryption key id=] to |keyId|.
+    1. Let (|requestBlob|, |context|) be the result of serializing |igMap| with |config| and
+      |igPAggCoordinatorMap|. The serialization method may follow that described in
+      [Section 2.2.4 of Bidding and Auction Services](https://privacysandbox.github.io/draft-ietf-bidding-and-auction-services/draft-ietf-bidding-and-auction-services.html#name-generating-a-request).
+    1. Let |result| be a new [=auction data per seller result=] with the following [=struct/items=]:
+        : [=auction data per seller result/seller=]
+        :: |seller|
+        : [=auction data per seller result/request=]
+        :: |requestBlob|
+    1. [=list/Append=] |result| to |results|.
+    1. Let |requestContext| be a new [=server auction request context=].
+    1. Set |requestContext|'s [=server auction request context/request ID=] field to |requestId|.
+    1. Set |requestContext|'s [=server auction request context/request context=] field to |context|.
+    1. [=map/Set=] |global|'s [=associated Document's=] [=node navigable's=]
+      [=traversable navigable's=] [=traversable navigable/saved Bidding and Auction request context=][|requestId|] to |requestContext|.

```

Comment:
this does not seem right, which keeps overwriting the requestId's context to the next config's context. Do we need to use |requestId| + |seller| as the key now? https://crsrc.org/c/content/browser/interest_group/ad_auction_page_data.h;drc=12855e28398f2355e8002664108acc3c289401e0;l=150

Reply:
Done. I forgot about that change, good catch!